transformers/modeling_tf_xlnet.py

```diff
@@ -135,7 +133,8 @@ def rel_attn_core(self, inputs, training=False):
 
         # position based attention score
         bd = tf.einsum('ibnd,jbnd->ijbn', q_head + self.r_r_bias, k_head_r)
-        bd = self.rel_shift(bd, klen=ac.shape[1])
+        ac_shape = tf.shape(ac)
```

Comment:
can you import `shape_list` from `modeling_tf_utils` and use it instead of `tf.shape`?
It is designed to work both in eager and non-eager modes.

cc @LysandreJik

Reply:
I have made the change and tested it!